File: JS/3340.check_balanced_string [1].js
/**
 * @param {string} num
 * @return {boolean}
 */
var isBalanced = function(num) {
    let sum0 = 0, sum1 = 0
    for(let i=0; i<num.length; i++){
        if(i % 2 === 0)
            sum0 += Number(num[i])
        else
            sum1 += Number(num[i])
    }

    return sum0 === sum1
};

File: CPP [3001-3500]/3340.check-balanced-string [1].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    bool isBalanced(string num) {
        int sum0 = 0, sum1 = 0;
        for(int i=0; i<num.length(); i++)
        {
            if(i % 2 == 0)
                sum0 += num[i] - '0';
            else
                sum1 += num[i] - '0';
        }

        return sum0 == sum1;
    }
};